lib/vmpooler/api/v1.rb

```diff
@@ -502,6 +572,26 @@ def invalid_templates(payload)
       invalid
     end
 
+    def invalid_template_or_size(payload)
+      invalid = []
+      payload.each do |pool, size|
+        invalid << pool unless pool_exists?(pool)
+        Integer(size) rescue invalid << pool
```

Comment:
Should this check for `size <= 0`?

Reply:
Yes, it probably should. I've pushed a change and tests that validate this.